starlite/middleware/session.py

```diff
@@ -0,0 +1,217 @@
+        self.aesgcm = AESGCM(config.secret.get_secret_value())
+
+    def dump_data(self, data: Any) -> List[bytes]:
+        """
+        Given an arbitrary data, dump it into a bytes string, encrypt, encode and split it into chunks of the desirable size.
+        Args:
+            data: Datum to dump
+
+        Returns:
+            List of encoded bytes string of a maximum length equal to the 'CHUNK_SIZE' constant.
+
+        """
+        serialized = dumps(data)
+        nonce = urandom(NONCE_SIZE)
+        encrypted = self.aesgcm.encrypt(nonce, nonce + serialized)
+        encoded = b64encode(encrypted)
+
+        if len(encoded) <= CHUNK_SIZE:
+            return [encoded]
+
+        result: List[bytes] = []
+        for i in range(0, len(encrypted), CHUNK_SIZE):
+            result.append(encrypted[i : i + CHUNK_SIZE])
+
+        return result
+
+    def load_data(self, data: List[str]) -> Any:
+        """Given a list of strings, decodes them into the session object.
+
+        Args:
+            data: A list of strings derived from the request's session cookie(s).
+
+        Returns:
+            A deserialized session value.
+        """
+        decoded = b64decode("".join(data))
+        nonce = decoded[:NONCE_SIZE]
+        decrypted = self.aesgcm.decrypt(nonce, decoded)
+        return loads(decrypted)
+
+    def create_send_wrapper(
+        self, scope: "Scope", send: "Send", should_vacate_session: bool
+    ) -> Callable[["Message"], Awaitable[None]]:
+        """
+        Creates a wrapper for the ASGI send function, which handles setting the cookies on the outgoing response.
+        Args:
+            scope: The ASGI connection scope.
+            send: The ASGI send function.
+            should_vacate_session: A boolean flag dictating whether the session cookie should be vacated on the client
+                side.
+
+        Returns:
+            None.
+        """
+
+        async def wrapped_send(message: "Message") -> None:
+            """A wrapper around the send function, declared in local scope to
+            use closure values.
+
+            Args:
+                message: An ASGI message.
+
+            Returns:
+                None
+            """
+            if message["type"] == "http.response.start":
+                headers = MutableHeaders(scope=message)
+                if should_vacate_session:
+                    cookie_params = self.config.dict(exclude_none=True, exclude={"secret", "max_age"})
+                    headers.append("Set-Cookie", Cookie(value="null", expires=0, **cookie_params).to_header())
+                else:
+                    data = self.dump_data(scope.get("session"))
+                    cookie_params = self.config.dict(exclude_none=True, exclude={"secret"})
+                    if len(data) == 1:
+                        cookie_params.pop("key")
+                        headers.append("Set-Cookie", Cookie(value=data[0].decode("utf-8"), **cookie_params).to_header())
```

Comment:
Missing `key` argument for `Cookie` class.

```python
headers.append("Set-Cookie", Cookie(key=self.config.key, value=data[0].decode("utf-8"), **cookie_params).to_header())

Reply:
done